### DISCUSSION TITLE
Is there a way to End/Close a Trace ? i.e, not allow/wait for anymore spans to get created under it?

### DISCUSSION THREAD

**Discussion by Suedo** [open-telemetry/opentelemetry-java-instrumentation#4181] [Q&A]
If you see the below image of a Trace timeline, the trace stays open and empty long after the microservices flow has ended.
Note: I am using OTEL java instrumentation to generate the traces, and feeding them to a Jaegar Backend

![image](https://user-images.githubusercontent.com/4963779/134149906-2f18c921-0036-417a-8369-c9e3f9c71436.png)

Is there a way for us to close the Trace once we understand that an execution flow is complete and Jaegar/OpenTelemetry should not wait for any more spans to be generated?

**Comment by iNikem** [comment]
There is no such thing as "end of trace". Trace is just a collection of spans. If Jaeger Backend shows timeline beyond the end of the last span, that is its bug.

**Reply by Suedo** [reply]
But how is it decided which is the 'last' span?  

One instrumented Microservice can be waiting for a return value from a non instrumented 3rd party service, as a result, there will a visible gap in the trace timeline.

But when we know there is nothing else to return/process, don't we need to stop the Trace ?  Or is there a way to say 'this is the last span' and thus end the trace?

**Reply by iNikem** [reply]
Nobody decides about "last span". Trace timeline should end on the end timestamp of the last known span. There is no other timestamp that can qualify as "end of trace". There is no such thing as "end of trace".

**Reply by trask** [reply]
@Suedo it sounds like possibly you are seeing the context leaking into unrelated code that's not part of the request/response lifecycle?

can you narrow down and explain in a bit more detail the boundary in that large trace between things you think should be part of the request/response lifecycle and things you think should not be?
